src/plugins/data/common/search/expressions/utils/courier_inspector_stats.ts

```diff
@@ -74,17 +74,26 @@ export function getResponseInspectorStats(
     };
   }
 
-  if (resp && resp.hits) {
+  if (resp && resp.hits?.total !== undefined) {
+    let value: string | undefined;
+    if (typeof resp.hits.total === 'number') {
```

Comment:
Can we just add a `TODO` here to remind us to remove this logic when `legacyHitsTotal` is removed?